src/hooks/useTimeTicks.ts

```diff
@@ -0,0 +1,98 @@
+import type { ScaleTime } from 'd3-scale';
+import { MutableRefObject, useEffect, useMemo, useState } from 'react';
+
+import { textDimensions } from '../utils';
+
+type Directions = 'horizontal' | 'vertical';
+
+export interface TimeTicks {
+  label: string;
+  position: number;
+  value: Date;
+}
+
+interface Options {
+  tickFormat?: (d: Date) => string;
+  minSpace?: number;
+}
+
+const TEST_HEIGHT = '+1234567890';
+
+export function useTimeTicks<Scale extends ScaleTime<number, number>>(
+  scale: Scale,
+  direction: Directions,
+  ref: MutableRefObject<SVGGElement | null>,
+  options: Options = {},
+): TimeTicks[] {
+  const [ticks, setTicks] = useState<Date[]>([]);
+
+  const range = scale.range();
+  if (!range) throw new Error('Range needs to be specified');
+
+  const domain = scale.domain();
+  if (!domain) throw new Error('Domain needs to be specified');
+
+  const { minSpace = 8 } = options;
+  const format = options?.tickFormat;
+  const tickFormat = useMemo(
+    () => (format ? (x: Date) => format(x) : scale.tickFormat()),
+    [format, scale],
+  );
+
+  const axisLength = Math.abs(range[0] - range[1]);
+
+  // Calculates the tick number that fits in the given space
+  useEffect(() => {
```

Comment:
The code in this `useEffect` is copy-pasted from existing code no? Could you refactor it so that all hooks use the same base it's going to be hard to maintain otherwise. It could be implemented as a hook.